packages/runtime-core/src/componentProps.ts

```diff
@@ -37,17 +37,17 @@ export type ComponentPropsOptions<P = Data> =
   | string[]
 
 export type ComponentObjectPropsOptions<P = Data> = {
-  [K in keyof P]: Prop<P[K]> | null
+  [K in keyof P]: Prop<P[K], P[K]> | null
 }
 
-export type Prop<T> = PropOptions<T> | PropType<T>
+export type Prop<T, D = any> = PropOptions<T, D> | PropType<T>
```

Comment:
```suggestion
export type Prop<T, D = T> = PropOptions<T, D> | PropType<T>
```

Default to `T` instead of `any`